kani-compiler/src/kani_middle/transform/check_uninit/mod.rs

```diff
@@ -465,36 +522,43 @@ impl<'a> UninitInstrumenter<'a> {
 /// will have the following byte mask `[true, true, true, false]`.
 pub fn mk_layout_operand(
     body: &mut MutableBody,
+    statements: &mut Vec<Statement>,
     source: &mut SourceInstruction,
-    position: InsertPosition,
     layout_byte_mask: &[bool],
 ) -> Operand {
-    Operand::Move(Place {
-        local: body.insert_assignment(
-            Rvalue::Aggregate(
-                AggregateKind::Array(Ty::bool_ty()),
-                layout_byte_mask
-                    .iter()
-                    .map(|byte| {
-                        Operand::Constant(ConstOperand {
-                            span: source.span(body.blocks()),
-                            user_ty: None,
-                            const_: MirConst::from_bool(*byte),
-                        })
-                    })
-                    .collect(),
-            ),
-            source,
-            position,
-        ),
-        projection: vec![],
-    })
+    let span = source.span(body.blocks());
+    let rvalue = Rvalue::Aggregate(
+        AggregateKind::Array(Ty::bool_ty()),
+        layout_byte_mask
+            .iter()
+            .map(|byte| {
+                Operand::Constant(ConstOperand {
+                    span: source.span(body.blocks()),
+                    user_ty: None,
+                    const_: MirConst::from_bool(*byte),
+                })
+            })
+            .collect(),
+    );
+    let ret_ty = rvalue.ty(body.locals()).unwrap();
+    let result = body.new_local(ret_ty, span, Mutability::Not);
+    let stmt = Statement { kind: StatementKind::Assign(Place::from(result), rvalue), span };
+    statements.push(stmt);
+
+    Operand::Move(Place { local: result, projection: vec![] })
 }
 
 /// If injecting a new call to the function before the current statement, need to skip the original
 /// statement when analyzing it as a part of the new basic block.
-fn collect_skipped(operation: &MemoryInitOp, body: &MutableBody, skip_first: &mut HashSet<usize>) {
-    if operation.position() == InsertPosition::Before {
+fn collect_skipped(
+    position: InsertPosition,
+    source: &SourceInstruction,
+    body: &MutableBody,
+    skip_first: &mut HashSet<usize>,
```

Comment:
shouldn't this be `HashSet<BasicBlockIdx>`?

Reply:
We were able to get rid of it altogether! 😁